dialogue-core/src/main/java/com/palantir/dialogue/core/HostAndLimitedChannel.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.palantir.dialogue.DialogueImmutablesStyle;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@DialogueImmutablesStyle
+interface HostAndLimitedChannel {
```

Comment:
Just realized this as an immutable marker makes it hard to implement the executed-on marking channel. Just going to have a look at what I can do

Reply:
Done

Reply:
I'm not sure I follow why this is -- can you elaborate? I found the interface a bit harder to follow.

Reply:
Reverted, host marking will live in NodeSelectionChannel